app/apis/leaderboards.js

```diff
@@ -56,26 +65,30 @@ async function mbp() {
             FROM Bookmarks GROUP BY pid
         ) A
         ORDER BY value DESC, pid ASC
-        LIMIT 10`
+        LIMIT 10
+        OFFSET $1 * 10`,
+        [page]
     );
     return res.rows;
 }
 
 export async function getLeaderboards(req, res) {
+    const page = (req.query.page ?? 1) - 1;
+    if (page < 0) return res.status(400).json({ messages: "Invalid page number" });
```

Comment:
does this handle if page number is too high? like 1000?

Reply:
It will just return an empty list. In the future will also have a max page # to give to client, so theoretically shouldn't get an invalidly large number, except if someone outside uses the API.